include/ocpp/common/evse_security_impl.hpp

```diff
@@ -48,17 +48,20 @@ class EvseSecurityImpl : public EvseSecurity {
     void update_ocsp_cache(const CertificateHashDataType& certificate_hash_data,
                            const std::string& ocsp_response) override;
     bool is_ca_certificate_installed(const CaCertificateType& certificate_type) override;
-    std::string generate_certificate_signing_request(const CertificateSigningUseEnum& certificate_type,
-                                                     const std::string& country, const std::string& organization,
-                                                     const std::string& common, bool use_tpm) override;
-    std::optional<KeyPair> get_key_pair(const CertificateSigningUseEnum& certificate_type) override;
+    GetCertificateSignRequestResult
+    generate_certificate_signing_request(const CertificateSigningUseEnum& certificate_type, const std::string& country,
+                                         const std::string& organization, const std::string& common,
+                                         bool use_tpm) override;
+    std::optional<CertificateInfo> get_leaf_certificate_info(const CertificateSigningUseEnum& certificate_type,
```

Comment:
So far we have mirrored the yaml interface from everest-core so I would like to stick to it and use `GetCertificateInfoResult` as the return type here.

Reply:
Understood, will modify for all interfaces.